Adding Rich Text Paragraph Formatting (including bullet points)

ST-Imrie opened the pull request:
I added a new function `RichTextParagraph` that supports adding styles at the paragraph level, which allows for styling such as bullet points, numbered lists, test alignment etc.

Changes:
- New RichTextParagraph Object
- Changed Jinja2 tages to `{{rr <var> }}` for inline rich text and `{{rp <var> }}` for paragraph level rich text.
- Added new test documents
- Minor changes to template.py 

Example shown below:
![image](https://github.com/user-attachments/assets/3d6b0e1f-a57f-4079-bf11-caa5343fd4a3)

Please let me know what you think

elapouya commented:
This is a good idea, but you cannot change `{{r xx}}` syntax to  `{{rr xx}}` for compatibility reason.

ST-Imrie commented:
That makes sense, this has no backwards compatibility. 

If I were to rewrite it with the tag `{{r <tag> }}` for inline (same as current so will be backwards compatible) and use maybe `{{q <tag> }}` (or any unused letter) for the paragraph level richtext, would that work instead?

elapouya commented:
`{{q <var>}}` will be fine. Could you give some more possible paragraph properties that can be used as it is not obvious for developers ?

ST-Imrie commented:
I've updated it to the r/q format and added some more examples to the template so it now looks like this. Is this okay or is there anything you'd like me to change?

![image](https://github.com/user-attachments/assets/2a4aa9ae-02c2-4c42-831b-ca6055ba949d)

elapouya commented:
testing...

elapouya commented:
Reading more precisely the code : why not using `{{p <var>}}` ? 
it already exists and remove also everything in paragraph, including the paragraph tags themselves : I tested this way, it worked...

elapouya commented:
@ST-Imrie : For information, I merged your code without the `{{q <var>}}` part as `{{p <var>}}` works well.
